torch/distributed/_tensor/dispatch.py

```diff
@@ -120,6 +122,14 @@ def operator_dispatch(
     kwargs: Dict[str, object],
     sharding_propagator: ShardingPropagator,
 ) -> object:
+    # Redistribute grad_output tensor to the same placement as input tensor
+    args = list(args)
+    if op_call == torch.ops.aten.convolution_backward.default:
+        assert isinstance(args[0], dtensor.DTensor) and isinstance(
+            args[1], dtensor.DTensor
+        )
+        args[0] = args[0].redistribute(args[1].device_mesh, args[1].placements)
+    args = tuple(args)
```

Comment:
@wanchaol shouldn't this resharding happen in `_operator_dispatch`?

Reply:
I have moved it to `convolution_backward_handler` to avoid this pollution.